src/app/screens/Home/index.tsx

```diff
@@ -1,480 +1,100 @@
-import {
-  CaretLeftIcon,
-  SendIcon,
-  ReceiveIcon,
-} from "@bitcoin-design/bitcoin-icons-react/filled";
-import AllowanceMenu from "@components/AllowanceMenu";
-import Button from "@components/Button";
-import Header from "@components/Header";
-import IconButton from "@components/IconButton";
-import Loading from "@components/Loading";
-import Progressbar from "@components/Progressbar";
-import PublisherCard from "@components/PublisherCard";
-import TransactionsTable from "@components/TransactionsTable";
-import { Tab } from "@headlessui/react";
-import dayjs from "dayjs";
-import relativeTime from "dayjs/plugin/relativeTime";
-import { useState, useEffect, useCallback } from "react";
-import { Trans, useTranslation } from "react-i18next";
-import { useNavigate } from "react-router-dom";
-import { toast } from "react-toastify";
+import { FC, useState, useEffect, useCallback } from "react";
 import browser from "webextension-polyfill";
-import { useSettings } from "~/app/context/SettingsContext";
-import { classNames } from "~/app/utils/index";
 import api from "~/common/lib/api";
-import lnurlLib from "~/common/lib/lnurl";
-import utils from "~/common/lib/utils";
-import { isLNURLDetailsError } from "~/common/utils/typeHelpers";
-import type { Allowance, Battery, Transaction } from "~/types";
+import type { Allowance, Battery } from "~/types";
 
-dayjs.extend(relativeTime);
-
-function Home() {
-  const {
-    isLoading: isLoadingSettings,
-    settings,
-    getFiatValue,
-  } = useSettings();
+import AllowanceView from "./AllowanceView";
+import DefaultView from "./DefaultView";
 
+const Home: FC = () => {
   const [allowance, setAllowance] = useState<Allowance | null>(null);
-  const [isBlocked, setIsBlocked] = useState<boolean>(false);
   const [currentUrl, setCurrentUrl] = useState<URL | null>(null);
-  const [payments, setPayments] = useState<Transaction[]>([]);
-  const [incomingTransactions, setIncomingTransactions] = useState<
-    Transaction[] | null
-  >(null);
   const [loadingAllowance, setLoadingAllowance] = useState(true);
-  const [loadingPayments, setLoadingPayments] = useState(true);
-  const [loadingSendSats, setLoadingSendSats] = useState(false);
-  const [loadingInvoices, setLoadingInvoices] = useState(false);
   const [lnData, setLnData] = useState<Battery[]>([]);
-  const navigate = useNavigate();
-  const { t } = useTranslation("translation", { keyPrefix: "home" });
-  const { t: tCommon } = useTranslation("common");
 
-  async function loadAllowance() {
+  const loadAllowance = useCallback(async () => {
     try {
-      const tabs = await browser.tabs.query({
-        active: true,
-        currentWindow: true,
-      });
-      const [currentTab] = tabs;
-      const url = new URL(currentTab.url as string);
-      setCurrentUrl(url);
-      const result = await api.getAllowance(url.host);
+      setLoadingAllowance(true);
+
+      if (!currentUrl) throw new Error("No established browser connection");
```

Comment:
what is this about? what means "No established browser connection" ? 

isn't also loadAllowance only called if there is currentUrl?

Reply:
This is for typescript. I added a comment to make this more clear